mmedit/models/editors/cyclegan/cyclegan.py

```diff
@@ -282,10 +282,10 @@ def val_step(self, data: dict) -> SampleList:
 
             for src_domain in self._reachable_domains:
                 target_domain = self.get_other_domains(src_domain)[0]
-                setattr(gen_sample, f'gt_{src_domain}',
-                        PixelData(data=inputs_dict[f'img_{src_domain}'][idx]))
-                setattr(gen_sample, f'fake_{src_domain}',
-                        PixelData(data=outputs[f'img_{src_domain}'][idx]))
+                fake_img = outputs[f'img_{target_domain}'][idx]
+                fake_img = self.data_preprocessor.destruct(
+                    fake_img, data_samples[idx], f'img_{target_domain}')
+                setattr(gen_sample, f'fake_{target_domain}', fake_img)
```

Comment:
I think using `setattr` to add attributes in `gen_sample` can be a bit hacky. 
I suggest using 
`gen_sample.set_field(fake_img, f'fake_{target_domain}', dtype=tensor)`
or 
`gen_sample.set_tensor_data(f'fake_{target_domain}', fake_img)`